File: ios/quadpay-merchant-sdk-react-native/quadpay-merchant-sdk-react-native/RNQuadPayPaymentWidget.m
//
//  RNQuadPayPaymentWidget.m
//  quadpay-merchant-sdk-react-native
//
//  Created by Petros Andreou on 25/01/2023.
//

#import "RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RNQuadPayPaymentWidgetManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(amount, NSString)
RCT_EXPORT_VIEW_PROPERTY(merchantId, NSString)
RCT_EXPORT_VIEW_PROPERTY(timelineColor, NSString)
RCT_EXPORT_VIEW_PROPERTY(hideHeader, NSString)
RCT_EXPORT_VIEW_PROPERTY(hideSubtitle, NSString)
RCT_EXPORT_VIEW_PROPERTY(hideTimeline, NSString)
RCT_EXPORT_VIEW_PROPERTY(isMFPPMerchant, NSString)
RCT_EXPORT_VIEW_PROPERTY(learnMoreUrl, NSString)
RCT_EXPORT_VIEW_PROPERTY(minModal, NSString)
@end

File: ios/quadpay-merchant-sdk-react-native/quadpay-merchant-sdk-react-native/RNQuadPayPaymentWidget.swift
//
//  RNQuadPayPaymentWidget.swift
//  quadpay-merchant-sdk-react-native
//
//  Created by Petros Andreou on 25/01/2023.
//

import Foundation
import QuadPaySDK

@objc(RNQuadPayPaymentWidgetManager)
class RNQuadPayPaymentWidgetManager: RCTViewManager{

    @objc var quadpayPaymentWidget = RNPaymentWidget()
   
    override func view() -> UIView!{
        return quadpayPaymentWidget
    }

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
